### PR TITLE
feat(WAF): add datasource to query the list of the dedicated domains

### PR DESCRIPTION
**Pull request by MrLiangSir**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccDatasourceWAFDedicatedDomains_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccDatasourceWAFDedicatedDomains_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceWAFDedicatedDomains_basic
=== PAUSE TestAccDatasourceWAFDedicatedDomains_basic
=== CONT  TestAccDatasourceWAFDedicatedDomains_basic
--- PASS: TestAccDatasourceWAFDedicatedDomains_basic (389.32s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       389.357s

### CODE REVIEW

File: docs/data-sources/waf_dedicated_domains.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The data source ID.
+
+* `domains` - A list of WAF dedicated domains.
+
+The `domains` block supports:
+
+* `id` - The ID of WAF dedicated domains.
```
Comment (deer-hang) [inline docs/data-sources/waf_dedicated_domains.md]:
```suggestion
* `id` - The ID of WAF dedicated domain.
```

File: docs/data-sources/waf_dedicated_domains.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+* `ipv6_enable` - Whether to support IPv6.
+
+* `description` - The description of WAF dedicated domain.
+
+* `policy_id` - The policy ID associated with the domain.
+
+* `protect_status` - The protection status of domain.
+
+* `access_status` - Whether a domain name is connected to WAF.
```
Comment (deer-hang) [inline docs/data-sources/waf_dedicated_domains.md]:
```
* `protect_status` - (Optional, Int) The protection status of domain, `0`: suspended, `1`: enabled.

* `access_status` - Whether a domain name is connected to WAF. Valid values are:
  + `0` - The domain name is not connected to WAF,
  + `1` - The domain name is connected to WAF.
```

File: huaweicloud/services/waf/data_source_huaweicloud_waf_dedicated_domains.go [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+			"is_dual_az":            utils.PathSearch("is_dual_az", v, nil),
+			"ipv6_enable":           utils.PathSearch("ipv6", v, nil),
+		})
+	}
+	return rst
+}
+
+func buildWAFDedicatedDomainsQueryParams(d *schema.ResourceData) string {
+	res := ""
+	if v, ok := d.GetOk("enterprise_project_id"); ok {
```
Comment (deer-hang) [inline huaweicloud/services/waf/data_source_huaweicloud_waf_dedicated_domains.go]: use `conf.GetEnterpriseProjectID()`

File: huaweicloud/services/waf/data_source_huaweicloud_waf_dedicated_domains.go [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"domain": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"enterprise_project_id": {
```
Comment (deer-hang) [inline huaweicloud/services/waf/data_source_huaweicloud_waf_dedicated_domains.go:66]: There is no such field in the response body.

File: huaweicloud/services/waf/data_source_huaweicloud_waf_dedicated_domains.go [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+		rst = append(rst, map[string]interface{}{
+			"id":                    utils.PathSearch("id", v, nil),
+			"domain":                utils.PathSearch("hostname", v, nil),
+			"policy_id":             utils.PathSearch("policyid", v, nil),
+			"enterprise_project_id": utils.PathSearch("enterprise_project_id", v, nil),
+			"description":           utils.PathSearch("description", v, nil),
+			"protect_status":        utils.PathSearch("protect_status", v, nil),
+			"website_name":          utils.PathSearch("web_tag", v, nil),
+			"access_status":         utils.PathSearch("access_status", v, nil),
+			"pci_3ds":               utils.PathSearch("pci_3ds", v, nil),
```
Comment (deer-hang) [inline huaweicloud/services/waf/data_source_huaweicloud_waf_dedicated_domains.go]:
```suggestion
			"pci_3ds":               utils.PathSearch("flag.pci_3ds", v, nil),
```
same question to field `pci_dss`, `is_dual_az`, `ipv6_enable`.

**Comment by MrLiangSir** [comment]
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccDatasourceWAFDedicatedDomains_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccDatasourceWAFDedicatedDomains_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceWAFDedicatedDomains_basic
=== PAUSE TestAccDatasourceWAFDedicatedDomains_basic
=== CONT  TestAccDatasourceWAFDedicatedDomains_basic
--- PASS: TestAccDatasourceWAFDedicatedDomains_basic (400.39s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       400.453s

**Comment by MrLiangSir** [comment]
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccDatasourceWAFDedicatedDomains_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccDatasourceWAFDedicatedDomains_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceWAFDedicatedDomains_basic
=== PAUSE TestAccDatasourceWAFDedicatedDomains_basic
=== CONT  TestAccDatasourceWAFDedicatedDomains_basic
--- PASS: TestAccDatasourceWAFDedicatedDomains_basic (369.53s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       369.571s

File: docs/data-sources/waf_dedicated_domains.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+* `ipv6_enable` - Whether to support IPv6.
+
+* `description` - The description of WAF dedicated domain.
+
+* `policy_id` - The policy ID associated with the domain.
+
+* `protect_status` - The protection status of domain,  `0`: suspended, `1`: enabled. Default value is `0`.
+
+* `access_status` - Whether a domain name is connected to WAF.Valid values are:
```
Comment (ShiChangkuo) [inline docs/data-sources/waf_dedicated_domains.md]: please add a space
```suggestion
* `access_status` - Whether a domain name is connected to WAF. Valid values are:
```

**Pull request merged**